plugins/variants/src/VcfAdapter/VcfAdapter.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  BaseFeatureDataAdapter,
+  BaseOptions,
+} from '@jbrowse/core/data_adapters/BaseAdapter'
+import { FileLocation, Region } from '@jbrowse/core/util/types'
+import { openLocation } from '@jbrowse/core/util/io'
+import { ObservableCreate } from '@jbrowse/core/util/rxjs'
+import { Feature } from '@jbrowse/core/util/simpleFeature'
+import { readConfObject } from '@jbrowse/core/configuration'
+import { AnyConfigurationModel } from '@jbrowse/core/configuration/configurationSchema'
+import VcfFeature from '../VcfTabixAdapter/VcfFeature'
+import VCF from '@gmod/vcf'
+import { unzip } from '@gmod/bgzf-filehandle'
+
+const readVcf = (f: string) => {
+  const lines = f.split('\n')
+  const header: string[] = []
+  const rest: string[] = []
+  lines.forEach(line => {
+    if (line.startsWith('#')) {
+      header.push(line)
+    } else if (line) {
+      rest.push(line)
+    }
+  })
+  return { header: header.join('\n'), lines: rest }
+}
+
+export default class VcfAdapter extends BaseFeatureDataAdapter {
+  public static capabilities = ['getFeatures', 'getRefNames']
+
+  private setupP?: Promise<Feature[]>
+
+  public constructor(config: AnyConfigurationModel) {
+    super(config)
+  }
+
+  public async getLines() {
+    const vcfLocation = readConfObject(
+      this.config,
+      'vcfLocation',
+    ) as FileLocation
+
+    let fileContents = (await openLocation(vcfLocation).readFile()) as string
```

Comment:
this may be an incorrect type assertion. the fileContents will actually be a node.js style "Buffer" here. if it called readFile('utf8') it would get a string but with no args it is a buffer (https://github.com/GMOD/generic-filehandle#async-readfileopts-options-promisebuffer--string kind of shows that it can return a buffer|string depending on args)

I think this also means that in the "else" case (where it doesn't unzip) it should call fileContents.toString() to get a string